test/system/255-auto-update.bats

```diff
@@ -89,91 +90,87 @@ function _wait_service_ready() {
     die "Timed out waiting for $sname to start"
 }
 
+# Wait for container to update, as confirmed by its image ID changing
 function _confirm_update() {
-    local sname=$1
-
-    local timeout=6
-    last_log=""
-    while [[ $timeout -gt 1 ]]; do
-        run journalctl -u $sname -n 10
-        if [[ "$output" == "$last_log" ]]; then
+    local cname=$1
+    local old_iid=$2
+
+    # Image has already been pulled, so this shouldn't take too long
+    local timeout=5
+    while [[ $timeout -gt 0 ]]; do
+        run_podman '?' inspect --format "{{.Image}}" $cname
+        if [[ $status != 0 ]]; then
+            if [[ $output =~ (no such object|does not exist in database): ]]; then
```

Comment:
This was the hardest part to get right. Sometimes it's one of those errors, sometimes the other.